src/debugpy/adapter/launchers.py

```diff
@@ -153,7 +153,7 @@ def on_launcher_connected(sock):
                 "env": env,
             }
             if cwd is not None:
-                request_args["cwd"] = cwd
+                request_args["cwd"] = compat.filename_str(cwd)
```

Comment:
I think this only ever comes up as non-Unicode if it's not specified but "program" is, and so the client request handler defaults to `os.path.dirname(program)`. It would be better for that handler to do the proper conversion - since it knows its origin, and thus knows the encoding - such that in this function, the argument is always `unicode` on entry. Basically the general policy for filenames is: convert to/from Unicode as early as possible at the boundary with non-Unicode OS / Python APIs, so that all code that's not at the boundary can just assume it's all Unicode.

Reply:
OK, I'll take a look at it (as a note, I'm already finishing for the day here, so, I'll take a look at this next week).

Reply:
I removed this... it actually didn't do anything.

On Python 2, everything in the `cmdline` and `cwd` are actually `str` (and not `unicode`).

i.e.: on `launch_request`, although the `args` and `cwd` are built from something as:

```
program = request("program", unicode)
args = [program]
cwd = request("cwd", unicode, optional=True)
``` 

Logging it with something as:
`log.info('program {0!r}({1}), cwd: {2!r}({3})', program, type(program), cwd, type(cwd))`

shows that the types are actually `str`:
`program 'C:/temp/รก/script.py'(<class 'str'>), cwd: 'X:/ptvsd_workspace/ptvsd'(<class 'str'>)`

-- note that the actual path I was debugging is: `C:\temp\á\script.py`, so, it seems to be encoded with `mbcs` already.

So, the `args` and `cwd` that actually arrive on `spawn_debuggee` are all `list[str]` (and the launch works due to that).

It seems that internally there are still things that are inconsistent on Python 2 (even though the launch works), but given that the code seems to work even with those inconsistencies and given that this is a Python 2 only issue, I'm leaving that untouched.

Reply:
Oops, my bad, it seems I got those logs on a python 3, not python 2... on Python 2 it does indeed show `unicode` (which seems to be correct).

Removing that line still works properly ;)